feature_engine/discretisation/geometric_width.py

```diff
@@ -0,0 +1,161 @@
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine._docstrings.fit_attributes import (
+    _binner_dict_docstring,
+    _feature_names_in_docstring,
+    _n_features_in_docstring,
+    _variables_attribute_docstring,
+)
+from feature_engine._docstrings.init_parameters.all_trasnformers import (
+    _variables_numerical_docstring,
+)
+from feature_engine._docstrings.init_parameters.discretisers import (
+    _precision_docstring,
+    _return_boundaries_docstring,
+    _return_object_docstring,
+)
+from feature_engine._docstrings.methods import (
+    _fit_discretiser_docstring,
+    _fit_transform_docstring,
+    _transform_discretiser_docstring,
+)
+from feature_engine._docstrings.substitute import Substitution
+from feature_engine._variable_handling.init_parameter_checks import (
+    _check_init_parameter_variables,
+)
+from feature_engine.discretisation.base_discretiser import BaseDiscretiser
+
+
+@Substitution(
+    return_object=_return_object_docstring,
+    return_boundaries=_return_boundaries_docstring,
+    precision=_precision_docstring,
+    binner_dict_=_binner_dict_docstring,
+    fit=_fit_discretiser_docstring,
+    transform=_transform_discretiser_docstring,
+    variables=_variables_numerical_docstring,
+    variables_=_variables_attribute_docstring,
+    feature_names_in_=_feature_names_in_docstring,
+    n_features_in_=_n_features_in_docstring,
+    fit_transform=_fit_transform_docstring,
+)
+class GeometricWidthDiscretiser(BaseDiscretiser):
+    """
+    The GeometricWidthDiscretiser() divides continuous numerical variables into
+    intervals of increasing width with equal increments. Note that the
+    proportion of observations per interval may vary.
+
+    Sizes (a) of n intervals will follow geometric progression
+
+    .. math::
+        a_i+1 = a_i r^(n+i)
+
+    The GeometricWidthDiscretiser() works only with numerical variables.
+    A list of variables can be passed as argument. Alternatively, the discretiser
+    will automatically select all numerical variables.
+
+    The GeometricWidthDiscretiser() first finds the boundaries for the intervals for
+    each variable. Then, it transforms the variables, that is, sorts the values into
+    the intervals.
+
+    Note: this discretiser needs higher precision, because bins width could be very
+    small. With smaller range of the binned values and/or higher number of bins,
+    smaller precision values could be needed.
```

Comment:
We need to re-word this paragraph a bit.

Reply:
How about this?

Note: The width of some bins might be very small. Thus, to allow this transformer to work properly, it might help to increase the precision value, that is, the number of decimal values allowed to define each bin. If the variable has a narrow range or you are sorting into several bins, allow greater precision (i.e., if precision = 3, then 0.001; if precision = 7, then 0.0001).